src/test/mockFactory.ts

```diff
@@ -22,8 +23,23 @@ export class MockFactory {
       noDeploy: null,
       region: null,
       stage: null,
-      watch: null
-    }
+      watch: null,
+    };
+  }
+
+  public static createTestArmSdkResponse<R>(model: any, statusCode: number): Promise<R> {
```

Comment:
Nice. Q: Deciding between `<T>` and `<R>`? What's the convention?

Reply:
`<T>` is typically used to reference some incoming types where as `<R>` is used for outgoing / return types.